pkg/auth/rhsso_authz_handler.go

```diff
@@ -176,6 +180,29 @@ func (a *AuthzHandler) HasAccessTo(ctx context.Context, obj interface{}, action
 	return false, errors.New("can not perform access check on this object")
 }
 
+func (a *AuthzHandler) HasOrgBasedCapability(ctx context.Context, capability string) (bool, error) {
+	if !a.isOrgBasedFunctionalityEnabled() {
+		return true, nil
+	}
+
+	orgID := ocm.OrgIDFromContext(ctx)
+	username := ocm.UserNameFromContext(ctx)
+	cacheKey := fmt.Sprintf("%s-%s-%s", orgID, capability, ocm.OrganizationCapabilityType)
+	if cacheData, existInCache := a.client.Cache.Get(cacheKey); existInCache {
+		isAllowed, ok := cacheData.(bool)
+		if !ok {
+			return false, errors.New("Expected from cache data to be from bool kind")
+		}
+		return isAllowed, nil
+	}
+
+	isAllowed, err := a.client.Authorization.CapabilityReview(context.Background(), fmt.Sprint(username), capability, ocm.OrganizationCapabilityType, orgID)
+	if shouldStorePayloadInCache(err) {
+		a.client.Cache.Set(cacheKey, isAllowed, 1*time.Minute)
```

Comment:
10 mins might be short enough (not sure if AMS also have a cache for capabilities, worth checking)

Reply:
yup. It does not.